packages/react/src/scroll-area/root/ScrollAreaRoot.tsx

```diff
@@ -178,13 +179,14 @@ export const ScrollAreaRoot = React.forwardRef(function ScrollAreaRoot(
     }
   });
 
-  function handlePointerEnterOrMove({ pointerType }: React.PointerEvent) {
-    const isTouch = pointerType === 'touch';
+  function handlePointerEnterOrMove(event: React.PointerEvent) {
+    const isTouch = event.pointerType === 'touch';
 
     setTouchModality(isTouch);
 
     if (!isTouch) {
-      setHovering(true);
+      const isTargetRootChild = rootRef.current?.contains(event.target as Element);
+      setHovering(!!isTargetRootChild);
```

Comment:
You can use the `contains` util for shadow DOM support, which accepts null values as well

Reply:
Nice, updated https://github.com/mui/base-ui/pull/2542/commits/5a73c5e182937f0b969869254f1ffb83cf4e7c72